pest-ion/tests/cli.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+use anyhow::Result;
+use assert_cmd::Command;
+use ion_rs::value::reader::*;
+use rstest::*;
+use std::fs::File;
+use std::io::{Read, Write};
+use tempfile::TempDir;
+
+/// The input or output mode of the CLI.
+enum FileMode {
+    /// Use `STDIN` or `STDOUT`
+    Default,
+    /// Use a named file.
+    Named,
+}
+
+#[rstest]
+#[case::simple(
+    r#"
+        a = { "a" ~ "b" }
+    "#,
+    r#"
+        {
+            a: {
+                type: normal,
+                expression: (sequence (string exact "a") (string exact "b")),
+            }
+        }
+    "#
+)]
+fn simple(
+    #[case] pest_src: &str,
+    #[case] ion_text: &str,
+    #[values("", "-t", "-p", "-b")] format_flag: &str,
+    #[values(FileMode::Default, FileMode::Named)] input_mode: FileMode,
+    #[values(FileMode::Default, FileMode::Named)] output_mode: FileMode,
+) -> Result<()> {
+    // working space for our tests when they need files
+    let temp_dir = TempDir::new()?;
+    let input_path = temp_dir.path().join("INPUT.pest");
+    let output_path = temp_dir.path().join("OUTPUT.ion");
+
+    let mut cmd = Command::cargo_bin("pest2ion")?;
+    if format_flag != "" {
+        cmd.arg(format_flag);
+    }
+    match output_mode {
+        FileMode::Default => {
+            // do nothing
+        }
+        FileMode::Named => {
+            // tell our driver to output to a file
+            cmd.arg("-o");
+            cmd.arg(&output_path);
+        }
+    }
+    match input_mode {
+        FileMode::Default => {
+            // do nothing
+        }
+        FileMode::Named => {
+            // dump our test data to input file
+            let mut input_file = File::create(&input_path)?;
+            input_file.write(pest_src.as_bytes())?;
+            input_file.flush()?;
+
+            // make this the input for our driver
+            cmd.arg(input_path.to_str().unwrap());
+        }
+    };
+    println!("{:?}", cmd);
```

Comment:
This was to help me visualize the permutations, it is visible if you run `cargo` with `--show-output`.

```
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-o" "/tmp/.tmpCJA5m0/OUTPUT.ion", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "/tmp/.tmpIZFcxJ/INPUT.pest", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-o" "/tmp/.tmpGNdUse/OUTPUT.ion" "/tmp/.tmpGNdUse/INPUT.pest", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-t", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-t" "-o" "/tmp/.tmpPM4bcq/OUTPUT.ion", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-t" "/tmp/.tmp1HNyMC/INPUT.pest", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-t" "-o" "/tmp/.tmpM8Nz6P/OUTPUT.ion" "/tmp/.tmpM8Nz6P/INPUT.pest", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-p", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-p" "-o" "/tmp/.tmpKpsoGP/OUTPUT.ion", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-p" "/tmp/.tmphjCwHT/INPUT.pest", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-p" "-o" "/tmp/.tmpZDh6Up/OUTPUT.ion" "/tmp/.tmpZDh6Up/INPUT.pest", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-b", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-b" "-o" "/tmp/.tmpXB9Mqf/OUTPUT.ion", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-b" "/tmp/.tmpcYHp3T/INPUT.pest", stdin: None, timeout: None }
-------------- TEST START --------------
Command { cmd: "/home/ANT.AMAZON.COM/almann/CLionProjects/partiql-lang-rust/target/debug/pest2ion" "-b" "-o" "/tmp/.tmpjHh3fL/OUTPUT.ion" "/tmp/.tmpjHh3fL/INPUT.pest", stdin: None, timeout: None }
```